Modules/Scripted/SampleData/SampleData.py

```diff
@@ -309,7 +318,14 @@ def downloadFromSource(self,source):
     for uri,fileName,nodeName in zip(source.uris,source.fileNames,source.nodeNames):
       filePath = self.downloadFileIntoCache(uri, fileName)
       if nodeName:
-        nodes.append(self.loadNode(filePath, nodeName, source.loadFileType, source.loadFileProperties))
+        loadedNode = self.loadNode(filePath, nodeName, source.loadFileType, source.loadFileProperties)
+        if loadedNode is None:
+          self.logMessage('<b>Load failed! Trying to download again...</b>', logging.ERROR)
+          file = qt.QFile(filePath)
+          if not file.remove():
+            self.logMessage('<b>Load failed! Unable to delete and try again loading %s!</b>' % filePath, logging.ERROR)
```

Comment:
If it failed to remove the file, I suggest to completely fail and not try again.